api/hasura/actions/_handlers/allocationCsv.ts

```diff
@@ -29,88 +32,120 @@ async function handler(req: VercelRequest, res: VercelResponse) {
       422
     );
   }
-  const { circles_by_pk } = await adminClient.query(
-    {
-      circles_by_pk: [
-        { id: circle_id },
-        {
-          fixed_payment_token_type: true,
-          users: [
-            {
-              where: {
-                _or: [
-                  {
-                    deleted_at: { _is_null: true },
-                  },
-                  {
-                    deleted_at: { _gt: epochObj.end_date },
-                  },
-                ],
-              },
-            },
-            {
-              id: true,
-              name: true,
-              address: true,
-              fixed_payment_amount: true,
-              received_gifts: [
-                { where: { epoch_id: { _eq: epochObj.id } } },
-                {
-                  tokens: true,
-                },
-              ],
-              sent_gifts: [
-                { where: { epoch_id: { _eq: epochObj.id } } },
-                {
-                  tokens: true,
-                },
-              ],
-            },
-          ],
-        },
-      ],
-    },
-    {
-      operationName: 'allocationCsv_getGifts',
-    }
-  );
-  assert(circles_by_pk, 'No Circle Found');
-  const fixedPaymentsEnabled =
-    isFeatureEnabled('fixed_payments') &&
-    circles_by_pk.fixed_payment_token_type;
-  const users = circles_by_pk.users;
   const grant = payload.grant ?? epochObj.grant;
+
   const totalTokensSent = epochObj.token_gifts.length
     ? epochObj.token_gifts
         .map(g => g.tokens)
         .reduce((total, tokens) => tokens + total)
     : 0;
-  const dateRange = `${formatShortDateTime(
-    epochObj.start_date
-  )} - ${formatShortDateTime(epochObj.end_date)}`.replace(/,/g, '');
+  const circle = await getCircleDetails(
+    circle_id,
+    epochObj.id,
+    epochObj.end_date
+  );
+  assert(circle, 'No Circle Found');
+  const fixedPaymentsEnabled =
+    isFeatureEnabled('fixed_payments') && !!circle.fixed_payment_token_type;
+
+  const userValues = generateCsvValues(
+    circle,
+    form_gift_amount,
+    gift_token_symbol,
+    totalTokensSent,
+    fixedPaymentsEnabled,
+    circle.fixed_payment_token_type,
```

Comment:
This doesnt seem like the best way to provide this value (from the circle details) since it can be changed by the user at any time. Why not use the same token if its a combined distribution, or use the vault symbol from the fixed distribution if its a separate distribution?

Reply:
i agree this is a real issue, and the root cause is that we don't take a snapshot of the circle's settings at the end of the epoch. 

> Why not use the same token if its a combined distribution, or use the vault symbol from the fixed distribution if its a separate distribution?

we can't assume there has been a distribution.